### DISCUSSION TITLE
Laravel Sail and Horizon infresh install not activate Horizon

### DISCUSSION THREAD

**Discussion by abkrim** [laravel/framework#43834] [Q&A]
When starting my sail app, Horizon es inactive. I need to start with `sail artisan horizon`

I see that docker proccess of sail has a `supervisord.conf`.

## Steps
- start sail project `sail up -d`
- Enter for check if supervisor is up (is up)

```bash
sail root-shell
root@fd84ad57e051:/var/www/html# ps aux|grep supervisor
root         1  0.1  0.3  32348 25956 ?        Ss   07:28   0:00 /usr/bin/python3 /usr/bin/supervisord -c /etc/supervisor/conf.d/supervisord.conf
root        39  0.0  0.0   2884  1312 pts/0    S+   07:30   0:00 grep --color=auto supervisor
```

But horizon is inactive. I  need to start in the machine with `sail artisan horizon`

I see that `supervisord.conf` call a `serve`

For me in production I use `command=php /path/mysite/artisan horizon`

I think this question is because is inactive.

Is possible autostart horizon in Sail?

Best regrads.

**Comment by dokicro** [comment]
First you have to copy `docker` folder to your project with `sail artisan sail:publish` in terminal.

After that in `dockeer/{VERSION}/supervisord.conf` add

```
[program:horizon]
command=/usr/bin/php /var/www/html/artisan horizon
autostart=true
autorestart=true
user=sail
redirect_stderr=true
stdout_logfile=/var/www/html/storage/logs/horizon.log
stopwaitsecs=360
```

After that run `sail build --no-cache` 

Thats it :)
